community/cypher/cypher-compiler-3.2/src/main/scala/org/neo4j/cypher/internal/compiler/v3_2/pipes/ExpandIntoPipe.scala

```diff
@@ -45,21 +45,32 @@ case class ExpandIntoPipe(source: Pipe,
   self =>
   private final val CACHE_SIZE = 100000
 
+  private val _relationships : ThreadLocal[Iterator[Relationship] with AutoCloseable] =
+    new ThreadLocal[Iterator[Relationship] with AutoCloseable]
```

Comment:
More `ThreadLocal` = more yuck. Again, it seems to me like having `internalCreateResults` return `Iterator[ExecutionContext] with AutoCloseable` is a better approach.